compiler/plc_xml/src/xml_parser/control.rs

```diff
@@ -0,0 +1,99 @@
+use ast::ast::{AstFactory, AstNode};
+use plc_diagnostics::diagnostics::Diagnostic;
+
+use crate::model::{
+    control::{Control, ControlKind},
+    fbd::{Node, NodeIndex},
+};
+
+use super::ParseSession;
+
+impl Control {
+    pub(crate) fn transform(&self, session: &ParseSession, index: &NodeIndex) -> Result<AstNode, Diagnostic> {
+        match self.kind {
+            ControlKind::Jump => unimplemented!(),
+            ControlKind::Label => unimplemented!(),
+            ControlKind::Return => transform_return(self, session, index),
+        }
+    }
+}
+
+fn transform_return(
+    control: &Control,
+    session: &ParseSession,
+    index: &NodeIndex,
+) -> Result<AstNode, Diagnostic> {
+    let Some(ref_local_id) = control.ref_local_id else {
+        let location = session.range_factory.create_block_location(control.local_id, None);
+        return Err(Diagnostic::empty_control_statement(location));
+    };
+
+    let Some(node) = index.get(&ref_local_id) else {
+        let location = session.range_factory.create_block_location(ref_local_id, None);
+        return Err(Diagnostic::undefined_node(control.local_id, ref_local_id, location));
+    };
+
+    let condition = match node {
+        Node::FunctionBlockVariable(variable) => Ok(variable.transform(session)),
+        Node::Block(block) => Ok(block.transform(session, index)),
+
+        _ => {
+            let location_control = session.range_factory.create_block_location(control.local_id, None);
```

Comment:
i'm sure the control can provide its exec order. and if it's none then it's none. 
I'll just change that in my commit with the jump / label